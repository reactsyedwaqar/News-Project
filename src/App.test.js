import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: jest
    .fn()
    .mockReturnValue({ environment: "dev", service: "fakeService" }),
}));

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
}));

test("App component snapshot", () => {
  const component = renderer.create(<App />);
  expect(component.toJSON()).toMatchSnapshot();
});
