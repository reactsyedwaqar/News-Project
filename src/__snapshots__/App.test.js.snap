// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div
      className="table-responsive-md"
    >
      <table
        cellPadding={5}
        cellSpacing={5}
        className="table table-striped table-hover"
      >
        <thead>
          <tr
            style={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "#ff742b",
                "color": "#fff",
                "fontSize": 16,
                "height": 20,
                "textAlign": "left",
                "whiteSpace": "normal",
              }
            }
          >
            <th
              style={
                Object {
                  "maxWidth": "20%",
                }
              }
            >
              Comments
            </th>
            <th
              style={
                Object {
                  "maxWidth": "20%",
                }
              }
            >
              Votes
            </th>
            <th
              style={
                Object {
                  "maxWidth": "20%",
                }
              }
            >
              UpVote
            </th>
            <th
              style={
                Object {
                  "maxWidth": "80%",
                }
              }
            >
              News Details
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              no data!
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      style={
        Object {
          "align": "right",
          "borderBottom": "3px solid #ff742b",
        }
      }
    >
      <span>
        Prev
      </span>
      <span
        style={
          Object {
            "marginLeft": 5,
            "marginRight": 5,
          }
        }
      >
         | 
      </span>
      <span>
        Next
      </span>
    </div>
    <br />
  </div>
</div>
`;
